### PR TITLE
fix(RN, JS): remove RN fragment, add callout for Auth.getPreferredMFA()

### PR DESCRIPTION
**Pull request by cwomack**
#### Description of changes:

Removes a RN fragment used on the [Auth MFA page](https://docs.amplify.aws/lib/auth/mfa/q/platform/js/) to consolidate to the JS fragment with JS/TS Code examples and adds callout for recommended process for using `Auth.getPreferredMFA()` to avoid unexpected responses from Cognito. 

#### Related GitHub issue #, if available:

Resolves Issue #5635 

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/auth/js/mfa.mdx [hunk 1]
```diff
@@ -203,6 +203,12 @@ try {
 </Block>
 </BlockSwitcher>
 
+<Callout warning>
+
+To avoid unexpected responses from Cognito, it is recommended to call the `Auth.setPreferredMFA()` API before retrieving MFA settings with `Auth.getPreferredMFA()`. If the MFA settings returned from Cognito do not align with the expected user’s preferences, trying to [setting the user MFA preferences](/lib/auth/mfa/q/platform/react-native/#setup-mfa-type) first.
```
Comment (abdallahshaban557) [inline src/fragments/lib/auth/js/mfa.mdx:208]: it is recommended to call the `Auth.setPreferredMFA()` when a user sets up their MFA method (or something similar).

Comment (cwomack) [reply src/fragments/lib/auth/js/mfa.mdx]: Updated the verbiage and condensed the callout.

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

**Review by israx — APPROVED** [review]
Thanks for putting this Chris! LGTM

**Review by katiegoines — APPROVED** [review]

**Pull request merged**